index.js

```diff
@@ -889,10 +889,8 @@ let createContainer = ({resolvers = [], dependencies = {}, factories, definition
 
         if (!definition.reuse) {
             destroyObject(instance, options);
-            definition.instance = null;
-        } else {
-            delete definition.instance;
         }
+        definition.instance = null;
```

Comment:
Also, if in session no one of reuse will be requested (A,B or C) instance will be destroyed by default process